google-cloud-pubsub/lib/google-cloud-pubsub.rb

```diff
@@ -143,4 +143,5 @@ def self.pubsub project_id = nil, credentials = nil, scope: nil,
   config.add_field! :emulator_host, default_emulator,
                     match: String, allow_nil: true
   config.add_field! :on_error, nil, match: Proc
+  config.add_field! :endpoint, nil, match: String
```

Comment:
What do you think about naming this `host`?

Reply:
The specific request was for the name `endpoint`, or `api_endpoint` as an alternative. This seems to be happening across languages (and across APIs).

Reply:
Can `endpoint` contain an optional port? The current GAPIC client has separate `SERVICE_ADDRESS` and `DEFAULT_SERVICE_PORT` values. But they are joined when passing to GRPC when creating a client. Do we want the config to expose both host and port separately? Or have one value containing both?

Reply:
The `emulator_host` value usually contains the port.

Reply:
This is an open question right now. I believe it should, but this is still being figured out for consistency across languages. So for now I think we should support hostname, leaving the option open to additionally recognize `host:port` later.